dpgen/generator/lib/lammps.py

```diff
@@ -167,6 +171,61 @@ def get_dumped_forces(
     ret = np.array(ret)
     return ret
 
+def get_all_dumped_forces(
+        file_name):
+    with open(file_name) as fp:        
+        lines = fp.read().split('\n')
+
+    ret = []
+    exist_natoms = False
+    exist_atoms = False
+
+    for idx,ii in enumerate(lines):
+
+        if 'ITEM: NUMBER OF ATOMS' in ii:
+            natoms = int(lines[idx+1])
+            exist_natoms = True
+
+        if 'ITEM: ATOMS' in ii:
+            keys = ii
+            keys = keys.replace('ITEM: ATOMS', '')
+            keys = keys.split()
+            idfx = keys.index('fx')
+            idfy = keys.index('fy')
+            idfz = keys.index('fz')
+            exist_atoms = True
+        
+            single_traj = []
+            for jj in range(idx+1, idx+natoms+1):
+                words = lines[jj].split()
+                single_traj.append([ float(words[jj]) for jj in [idfx, idfy, idfz] ])
+            single_traj = np.array(single_traj)
+            ret.append(single_traj)
+    
+    if exist_natoms is False:
+        raise RuntimeError('wrong dump file format, cannot find number of atoms', file_name)
+    if exist_atoms is False:
+        raise RuntimeError('wrong dump file format, cannot find dump keys', file_name)
+    return ret
+
+def generate_single_traj(all_traj, traj_ind, single_traj):
+    with open(all_traj) as all_traj_fp:        
+        lines = all_traj_fp.read().split('\n')    
```

Comment:
This looks pretty slow where the entire trajectory, size of which may be several GBs, is read into memory for n_tasks times.

Reply:
Agree, we should read the trajectory only once.